ratpack-kafka-consumer/src/main/java/smartthings/ratpack/kafka/Consumer.java

```diff
@@ -0,0 +1,57 @@
+package smartthings.ratpack.kafka;
+
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import ratpack.func.Action;
+
+import java.util.Properties;
+
+/**
+ * Interface for creating a Kafka Consumer.  Implementers will have their consume method called as messages become
+ * available.
+ * @param <K> Key deserialization class.
+ * @param <V> Value deserialization class.
+ */
+public interface Consumer<K, V> {
+	/**
+	 * Method will be called on successful Kafka poll with fetched messages.
+	 * @param records
+	 */
+	void consume(ConsumerRecords<K, V> records) throws Exception;
+
+	/**
+	 * Kafka topics in which to subscribe.
+	 * @return
+	 */
+	String[] getTopics();
+
+	/**
+	 * Kafka Consumer Group ID.
+	 * @return
+	 */
+	String getGroup();
+
+	/**
+	 * Determines how many consumers will be spun up on application start.
+	 * @return
+     */
+	default Integer getConcurrencyLevel() {
```

Comment:
I'm curious as to others thoughts.  I think I prefer the flexibility.  When you say 'single process' are you referring to the JVM?  In the case of the above that will trigger the KafkaConsumerService to launch X threads each w/ their own KafkaConsumer client.